packages/api/src/external/hie/register-facility.ts

```diff
@@ -0,0 +1,300 @@
+  }
+  const existingFacilityName = existingFacility.name?.value;
+  if (!existingFacilityName) throw new NotFoundError("CQ OBO organization has no name");
+  return existingFacilityName;
+}
+
+function createFacilityDetails(
+  cxId: string,
+  facility: FacilityDetails,
+  address: AddressWithCoordinates
+): FacilityCreate {
+  const addressStrict = removeCoordinates(address);
+
+  const facilityDetails = {
+    cxId,
+    data: {
+      name: facility.nameInMetriport,
+      npi: facility.npi,
+      address: addressStrict,
+    },
+    type: facility.type,
+  };
+
+  return facilityDetails;
+}
+
+function removeCoordinates(address: AddressWithCoordinates): AddressStrict {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { lat, lon, ...rest } = address;
+  return rest;
+}
+
+type CreateOrUpdateFacility =
+  | FacilityCreate
+  | (FacilityUpdate & {
+      cxId: never;
+      oid: never;
+      facilityNumber: never;
+      type: never;
+    });
+
+async function createOrUpdateFacility(
+  cxId: string,
+  facilityId: string | undefined,
+  facilityNpi: string,
+  facility: CreateOrUpdateFacility
+): Promise<FacilityModel> {
+  if (facilityId) {
+    await getFacilityStrictOrFail({ cxId, id: facilityId, npi: facilityNpi });
+    const updatedFacility = await updateFacility({
+      id: facilityId,
+      ...facility,
+    });
+
+    return updatedFacility;
+  }
+
+  const createdFacility = await createFacility(facility);
+
+  return createdFacility;
+}
+
+function buildCqOboOrgName(vendorName: string, orgName: string, oboOid?: string): string {
+  if (oboOid) {
+    return `${vendorName} - ${orgName} #OBO# ${oboOid}`;
+  }
+
+  return `${vendorName} - ${orgName}`;
+}
+
+function buildCwOboOrgName(vendorName: string, orgName: string, oboOid?: string): string {
+  if (oboOid) {
+    return `${vendorName} - ${orgName} -OBO- ${oboOid}`;
+  }
+  return `${vendorName} - ${orgName}`;
+}
+
+async function createOrUpdateInCq(
+  facility: FacilityModel,
+  cxOid: string,
+  orgName: string,
+  coordinates: AddressWithCoordinates
+) {
+  const { log } = out("createOrUpdateInCq");
+  const { address } = facility.data;
+
+  const addressLine = address.addressLine2
+    ? `${address.addressLine1}, ${address.addressLine2}`
+    : address.addressLine1;
+
+  log(`Creating/Updating a CQ entry with this OID ${facility.oid} and name ${orgName}`);
+
+  await createOrUpdateCQOrganization({
+    name: orgName,
+    addressLine1: addressLine,
+    lat: coordinates.lat,
+    lon: coordinates.lon,
+    city: address.city,
+    state: address.state,
+    postalCode: address.zip,
+    oid: facility.oid,
+    contactName: metriportCompanyDetails.name,
+    phone: metriportCompanyDetails.phone,
+    email: metriportEmailForCq,
+    parentOrgOid: cxOid,
+    role: "Connection" as const,
+  });
+}
+
+// WHY DO WE ONLY CREATE BUT WE CREATE OR UPDATE IN CQ?
```

Comment:
Left comment - is this an oversight? Or is there something im missing

Reply:
Does the create also update?

Reply:
It only creates, its mapped as an outstanding item of https://github.com/metriport/metriport-internal/issues/1706